src/Sarif/Baseline/V2/V2ResultMatcher.cs

```diff
@@ -194,6 +231,54 @@ private void LinkIfSimilar(int beforeIndex, int afterIndex)
             }
         }
 
+        private ExtractedResult FirstWithUri(ExtractedResult desiredUri, IList<ExtractedResult> set, ref int fromIndex)
+        {
+            // Find the first Result with a Uri *matching* the desired one, or null if there aren't any
+            for (; fromIndex < set.Count; ++fromIndex)
+            {
+                int whereCmp = WhereComparer.CompareFirstArtifactUri(set[fromIndex], desiredUri);
```

Comment:
>CompareFirstArtifactUri [](start = 45, length = 23)

I kind of "get" the decision to put this method in `WhereComparer` but I don't think it belongs there, based on the Single Responsibility Principle. `WhereComparer`'s sole responsibility is to decide if two `ExtractedResults` have the same location (for a particular definition of "same"). This method doesn't contribute to that responsibility. It _does_ contribute to this class's responsibility of matching two sets of results, so I think it belongs in this class.

Reply:
I understand your point, but disagree with this one. I want to encapsulate all of the ExtractedResult where property related extraction and comparison to one place and want to be able to reuse it if some other class wants these comparisons. So, I guess I have a broader definition of that the responsibility of WhereComparer is. I'm thinking, for example, that SarifToCsv might reuse this logic later to get the first artifact uri out for two Results, though then I'd have to resolve the existing conflict between Result and ExtractedResult.